### PR TITLE
Create In-Guidebook Errors

### PR DESCRIPTION
**Pull request by Aeshus**
## About the PR

This PR changes the error-handling for the Guidebook system.

Instead of errors just being dumped in the console with an ambiguous "ERROR: Failed to parse document", it now tries to isolate the errors into the area it was parsed from and provides an error message plus the thing it tried to parse. 

No player should be seeing these error messages. Instead, these are for making the developer experience better for guidebook writers.

Also I use ISawmill now because someone told me to.

## Why / Balance

When making large guidebook changes, it's common to occasionally make some kind of syntax error. 

The problem arises is that the guidebook just refuses to render and leaves an obtuse error message that you have to go sifting through to find the issue. You then have to repeat that until all of the errors are fixed.

This now tries to point out all of the issues inside the guidebook, letting you at least preview the rest of the entry and see all the errors at once.

## Technical details

There's a new GuidebookError UI that displays the original text (if it's available) along with the error message.

Basically I hook into all of the places that used to error and log to propagate up and replaced it with a new GuidebookError placeholder and still print the error. But now it continues to parse and render. 

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

The UI isn't final (surprise-surprise). I'm hoping someone can help me make it look better!

image

## Breaking changes

N/A

**Changelog**

No cl (should not be user-facing)

### CODE REVIEW

File: Content.Client/Guidebook/Controls/GuidebookError.xaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            <OutputPanel Margin="5" MinHeight="75" VerticalExpand="True" Name="Original">
+                <OutputPanel.StyleBoxOverride>
+                    <gfx:StyleBoxFlat BorderThickness="0 0 0 1" BorderColor="Gray"
+                                      ContentMarginLeftOverride="3" ContentMarginRightOverride="3"
+                                      ContentMarginBottomOverride="3" ContentMarginTopOverride="3" />
+                </OutputPanel.StyleBoxOverride>
+            </OutputPanel>
+
+            <Collapsible Margin="5" MinHeight="75" VerticalExpand="True">
+                <CollapsibleHeading Title="Error Message" />
```
Comment (metalgearsloth) [inline Content.Client/Guidebook/Controls/GuidebookError.xaml:27]: All client-facing text must be localised, including parsing error.

File: Content.Client/Guidebook/DocumentParsingManager.cs [hunk 2]
```diff
@@ -22,24 +24,29 @@ public sealed partial class DocumentParsingManager
...
+    private ISawmill _sawmill = default!;
+    private Parser<char, Control> _tagParser = default!;
     public Parser<char, IEnumerable<Control>> ControlParser = default!;
 
     public void Initialize()
     {
         _tagParser = TryOpeningTag
-            .Assert(_tagControlParsers.ContainsKey, tag => $"unknown tag: {tag}")
+            // .Assert(_tagControlParsers.ContainsKey, tag => $"unknown tag: {tag}")
             .Bind(tag => _tagControlParsers[tag]);
```
Comment (metalgearsloth) [inline Content.Client/Guidebook/DocumentParsingManager.cs:37]: Any particular reason this was commented out?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Client/Guidebook/Controls/GuidebookError.xaml (hunk 1 above)
Comment (Aeshus) [reply Content.Client/Guidebook/Controls/GuidebookError.xaml:27]: Should be fixed, thanks!

File: Content.Client/Guidebook/DocumentParsingManager.cs (hunk 2 above)
Comment (Aeshus) [reply Content.Client/Guidebook/DocumentParsingManager.cs]: I commented this out originally as I was trying to see if I could easily make it continue instead of stopping parsing at the first invalid tag error.

I had trouble implementing it so I didn't continue with the rest of the implementation.

This should be uncommented as while both (commented/uncommented) halt the parsing, the uncommented version gives a better error message.

**Review by metalgearsloth — APPROVED** [review]
